flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -313,6 +325,13 @@ public JobMaster(
 			.orElse(FutureUtils.completedExceptionally(new JobMasterException("The JobMaster has not been started with a REST endpoint.")));
 
 		this.metricQueryServicePath = metricQueryServicePath;
+		this.stackTraceSampleCoordinator = new StackTraceSampleCoordinator(rpcService.getExecutor(), rpcTimeout.toMilliseconds());
+		this.backPressureStatsTracker = new BackPressureStatsTracker(
```

Comment:
The `BackPressureStatsTracker` and the `StackTraceSampleCoordinator` could go into the `JobManagerServices`. That way, we would have a way to pass it into the `JobMaster`.

Reply:
done